File: backend/FinanceTracker.Api/Models/Transaction.cs
namespace FinanceTracker.Api.Models;

public class Transaction
{
    public Guid Id { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public TransactionType Type { get; set; }
    public DateOnly Date { get; set; }
    public Guid UserId { get; set; }
    public Guid CategoryId { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: frontend/src/types/transaction.ts
export type TransactionType = 'Income' | 'Expense'

export interface Transaction {
  id: string
  description: string
  amount: number
  type: TransactionType
  date: string
  categoryId: string
  categoryName: string
}

export interface TransactionFilter {
  from?: string
  to?: string
  categoryId?: string
}

export interface TransactionRequest {
  description: string
  amount: number
  type: TransactionType
  date: string
  categoryId: string
}
